spec/factories/digital_objects/items.rb

```diff
@@ -23,5 +23,11 @@
         digital_object.primary_project = create(:project)
       end
     end
+
+    trait :with_asset do
+      after(:create) do |digital_object|
+        digital_object.append_child_uid(create(:asset, parent: digital_object).uid)
```

Comment:
One concern about this method of adding an asset to the item factory object is that you've set up a single-directional relationship between these two objects when normally there would be a bidirectional relationship. Could lead to inconsistencies during testing.

Reply:
Does the asset factory use the parent delta to save the parent again? It might just be redundant.

Reply:
Yeah, I think the `append_child_uid` part is redundant.

You can try replacing:
`digital_object.append_child_uid(create(:asset, parent: digital_object).uid)`
with
`create(:asset, parent: digital_object)`

Reply:
Actually, then your actual factory-returned instance of the parent object wouldn't be reloaded to recognize the addition, so I guess it does make sense to keep it as is.

Reply:
We just need to ensure, long-term, that whatever parent adding logic happens inside of the digital object save method is mirrored here in the `with_asset` trait.  Hyacinth generally assumes that children add their parents, rather than the other way around.